ecs-agent/daemonimages/csidriver/driver/node_linux.go

```diff
@@ -0,0 +1,50 @@
+//go:build linux
+// +build linux
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package driver
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"golang.org/x/sys/unix"
+)
+
+// IsBlock checks if the given path is a block device
+func (d *nodeService) IsBlockDevice(fullPath string) (bool, error) {
+	var st unix.Stat_t
+	err := unix.Stat(fullPath, &st)
+	if err != nil {
+		return false, err
+	}
+
+	return (st.Mode & unix.S_IFMT) == unix.S_IFBLK, nil
+}
+
+func (d *nodeService) getBlockSizeBytes(devicePath string) (int64, error) {
+	cmd := d.mounter.(*NodeMounter).Exec.Command("blockdev", "--getsize64", devicePath)
```

Comment:
Also why can't we simply execute the command using [os/exec](https://pkg.go.dev/os/exec) package instead of indirectly doing that using `d.mounter.(*NodeMounter).Exec.Command`?

Reply:
The EBS team did an intentional refactor about this method `getBlockSizeBytes`: move this from `mounter` to `nodeService` - https://github.com/kubernetes-sigs/aws-ebs-csi-driver/commit/2874b2db9c83b9832e005426416d9cd14458be34
But I don't have the context about this neither.

Reply:
> Also why can't we simply execute the command using [os/exec](https://pkg.go.dev/os/exec) package instead of indirectly doing that using d.mounter.(*NodeMounter).Exec.Command?

Checked the k8s.io/utils 's implementation: it wraps the os.exec so it can capture errors.

> Maybe we could add a method to Mounter that exposes a [utilexec.Interface](https://pkg.go.dev/k8s.io/utils/exec#Interface)?

Looks like the EBS team prefer to not doing this per this commit - 
https://github.com/kubernetes-sigs/aws-ebs-csi-driver/commit/fbd9b4ee097e89fc61f46cbdc338b6670cf85e4f#diff-effb211911ba91cf53294998a2acd12e9976f272a4047c319caaded0374ad966L38

Reply:
I'll reach out to the owner to get more context.

Reply:
although we're implementing this ourselves, I'd prefer that we stick as closely as possible to the existing EBS CSI Driver's implementation so there's minimal drift if we were to ever use the off-the-shelf CSI driver.